### ISSUE TITLE
Navmesh for character not working

### ISSUE THREAD

**Issue by SpiderMack** [Aurasoft-UK/Skyline-Game-Engine#6]
A character with Lua script using navmesh to movedoesn't work.
The character doesn't follow navmesh to destination position.

**Comment by StarFireUK** [comment]
Have tried this code with a single tile navmesh and it works.
obj = 0;
move =1;
x, y, z =0;
stopped = true;

function onInit(objID)
	obj = objID;
	destination()
end

function postInit()
	navmesh.addEntity(obj);
	navmesh.showAgentDebug(obj, 1);
	navmesh.setAgentMaxSpeed(obj, 5);
end

function onUpdate(timeDelta)
	if( navmesh.isDestinationReached(obj) ==1 and stopped == false)then
		navmesh.setAgentControlled(obj,0);
		anim.playAnimation(obj, anim.getFromMap(obj, "Idle1"),30,1);
		stopped = true;
	end
end

function clickDestination()
	stopped = false;
	navmesh.setAgentControlled(obj,1);
	x,y,z = sky.raycastToCursor();
	navmesh.setDestination(obj, x, y, z);
	anim.playAnimation(obj, anim.getFromMap(obj, "Run"),30,1);
end

function onMouseDown( button, x, y )
	clickDestination()
end

**Comment by StarFireUK** [comment]
There is a few issues if the character moves too quick you may get an undesirable spinning. Also the destination debug sphere gets to the destination position but not the character. These are minor issues that will be resolved.

**Comment by StarFireUK** [comment]
Have found a reload scene crash with the navmesh!

**Comment by SpiderMack** [comment]
Your script works but only when the character has no physics.

If you add dynamic sphere rigibody or a simple character controller, the character no more goes where you point, or does not move.

Navmesh should work in combination with physics otherwise you can't make gameplay collisions detection, it would be almost useless.

**Comment by SolarPortal** [comment]
you cant use dynamic sphere rigibody, only an SCC can be used with that script.
Would you be able to record a video of the character, its properties and the play the game so we can see what you see. Post this to the forum with a link here :) Thank you.

**Comment by StarFireUK** [comment]
This script is working with the default female character with a SCC and a microscript. Character moves over undulated ground and up ramp.

**Comment by StarFireUK** [comment]
we have found that there is an error loading back a navmesh with type: "Single Navmesh", best to use tiled navmesh for now :)

**Comment by SpiderMack** [comment]
I tried tiled navmesh,  the zombie character with your script and a SCC.It never goes where i point.
Try to make the zombie working with SCC and navmesh script.

Unity work as components, navmesh component forces the whole stack of component to go where navmesh agent goes. So you can add any physics objects and collision you want , they will follow the navmesh when it moves.
It's simple, it works.
Why Skyline could not do the same later ?

**Comment by SolarPortal** [comment]
Couple of bug fixes done, currently not the movement, but:
* Calling isDestinationReached() in lua before setting a destination would crash.
* Loading a scene that used Single navmesh mode, would generate an incorrect navmesh.

**Comment by SolarPortal** [comment]
> "Try to make the zombie working with SCC and navmesh script."
Is this the standard Game Object zombie, as this has no navmesh code in the script....
Or are you using the base zombie mesh and building it from the ground up....?

**Comment by SpiderMack** [comment]
Yes this is the standard Zombie, i created a new Lua file with code you put here and replaced the zombie Lua script with this new one. It works the zombie goes where i point.

Once i add a SCC to the zombie, navmesh no more works as intended.
Test it and see by yourself.

**Comment by SolarPortal** [comment]
have you tried setting the zombie "SCC offset" to "0,-1.0, 0" which should sit the zombie at the floor?

**Comment by DevMagicLord** [comment]
SCC offset is a hack ?

Whatever zombie is on floor or not , in Unity all objects stack will follow the navmesh agent.
In Unity navmesh is not dependent on other components, it behaves as it must, other components just follow it's movement.
Why Skyline navmesh could not also be independent and always work whatever are the other components in properties ?
